lib/grpnames.gi

```diff
@@ -1619,6 +1619,11 @@ InstallMethod( StructureDescription,
            k,            # maximal power of d in Size(G)
            pi;           # subset of primes
 
+    if not HasIsFinite(G) and IsFinite(G) then
```

Comment:
Did you intentionally put this line here because ````IsFinite```` ranks higher than ````IsAbelian````? I guess an ````IsFinite```` check is not too much for infinite abelian groups. 

BTW, I could make two different methods for ````StructureDescription````, one for finite groups and one for abelian, but for that I would need to hack around with some of the internal functions, so that is why I rather did it this way. And I did the ````IsAbelian```` check first, because that is the first step of the finite method, as well, thus saving time on the infinite abelian groups, and not wasting any time on the others (unless abelian test is faster for groups that know they are finite).